make default aspect DropShadows instead of Inset for UbuntuShape

mymike00 opened the pull request:
this changes the default apsect of the UbuntuShape from Inset to DropShadow
**NOTE**: this will change the look of a load of components that uses the ubuntushape like buttons, switches, checkboxes, optionselectors and also the textfield in the messaging-app
I really like it but someone maybe won't. here are some screenshots

optionselector and checkboxes
![optionselector and checkboxes](https://user-images.githubusercontent.com/28359593/47028345-82bd1080-d169-11e8-8f24-c3659a5452ef.png)

switches
![switches](https://user-images.githubusercontent.com/28359593/47028434-b009be80-d169-11e8-99f8-1f991a2862d6.png)

buttons
![buttons](https://user-images.githubusercontent.com/28359593/47028538-dc253f80-d169-11e8-8019-ab5db9a1e376.png)

mariogrip committed:
Make sure both var is unsigned

mariogrip committed:
Make apicheck build on qt 5.8.2+

mariogrip committed:
Set the correct dir of headerdirs

mariogrip committed:
Build for bionic

mariogrip committed:
Workaround for silly bug

Will crap out with "Project ERROR: You cannot build examples inside the
Qt source tree, except as part of a proper Qt build." but running qmake
first without config seems to work

mariogrip committed:
Disable i18n type since QT disallows lowercase types

This does not effect i18n context prop, so i18n will work as normaly

https://github.com/qt/qtdeclarative/commit/1e350a8c98d9c98823dde83a6745d2f26a9c0785

mariogrip committed:
Merge pull request #8 from ubports/bionic_-_i18nfix

Disable i18n type since QT disallows lowercase types

mariogrip committed:
Set correct scaling on new spinners (https://github.com/ubports/ubuntu-touch/issues/568)

mymike00 committed:
imported pickers in the example code

Added import Ubuntu.Components.Pickers 1.3 in the example code
and changed MainWindow to MainView

mariogrip committed:
Revert "If GRID_UNIT_PX set, always use it."

This reverts commit 3f9d9ec030d796cf6d54657ceb6968e5021b5d2b.

mariogrip committed:
Revert "Do likewise in the other UCUnits constructor"

This reverts commit fc2c455b776916a2fe0880658e0231fb50610f16.

mariogrip committed:
If Qt HiDpi scaling is active, use that insted of GRID_UNIT_PX

This is the first move to depricate GRID_UNIT_PX all togheter

mariogrip committed:
Add back m_devicePixelRatio to make sure we dont make regression

devicePixelRatio should always be 1 if qts hidpi is disabled afaik, but lets add it back to be sure

mymike00 committed:
make default aspect DropShadows instead of Inset for UbuntuShape

mymike00 committed:
make default aspect DropShadows instead of Inset for UbuntuShape

mymike00 commented on src/UbuntuToolkit/ucubuntushape.cpp:
```diff
@@ -432,6 +432,7 @@ void UCUbuntuShape::setRadius(const QString& radius)
     \list
     \li \b UbuntuShape.Flat - no effects applied
     \li \b UbuntuShape.Inset - inner shadow slightly moved downwards and bevelled bottom
+    \li \b UbuntuShape.DropShadow - outer non-blurred shadow slightly moved downwards
```
is this description ok?

mymike00 commented on src/UbuntuToolkit/ucubuntushape.cpp:
```diff
@@ -857,7 +858,7 @@ void UCUbuntuShape::setBorderSource(const QString& borderSource)
         quint8 aspect;
         if (borderSource.endsWith(QStringLiteral("radius_idle.sci"))) {
-            aspect = Inset;
+            aspect = DropShadow;
```
not very very sure I have to change this also but I think it should...

nfsprodriver commented:
I love the new optic. It bring a even more modern flare to Ubuntu Touch.

balcy committed:
when determining the height of the keyboard rectangle, take Screen.devicePixelRatio into account
without that the popups are scrolled up too much, if the on screen keyboard is shown

cibersheep commented:
Interesting. I'm not a big fun if the Inset.
I like your changes for the buttons and textfields but not for the optionselectors and switches.

The positive aspect is that only changes the defaults but the downside is that we might want to change the behaviour on some elements.

mymike00 commented:
I like them also like this, but it wouldn't be so much effort changing also the components...

cibersheep commented:
Ok. Let's try this in the devel. If we feel like, we can revert it or modify some components before arrives RC.
Do you think?

mymike00 commented:
Yeah, let's merge!

cibersheep commented:
You got green light! Wweeeee... after OTA6 (which is «at the turn of the corner» meaning that will happen really soon).
So, if you don't mind waiting a little bit more, we will see your changes very soon.
THANKS!

UniversalSuperBox committed:
Move bzr config to git

UniversalSuperBox committed:
Revert "Make apicheck build on qt 5.8.2+"

mariogrip committed:
Change to markdown file

UniversalSuperBox committed:
Refactor apicheck to use QQmltype scalar

We were seeing issues with using apicheck on Qt 5.9.5. Really strange
memory issues leading to segfaults. This commit refactors apicheck
to use the scalar QQmlType everywhere and fix any compile issues
that cropped up.

UniversalSuperBox committed:
Update README and control to specify Qt 5.9

My changes to apicheck in
d85ab399d0a97d035ad473aa4449f631151e6566
make us incompatible with anything less than Qt 5.9

UniversalSuperBox committed:
Register i18n as I18n for apicheck

UniversalSuperBox committed:
Run tests with multiple cores

UniversalSuperBox committed:
Replace qmlInfo with qmlWarning to retain behavior

cibersheep commented:
So, has it arrive the time to try to merge this?
I see there's an error in the building tests...

mariogrip committed:
Set current dir as HOME

UniversalSuperBox committed:
You can't assign null to 'property QtObject'

UniversalSuperBox committed:
Remove inversemousearea tests which test nothing

UniversalSuperBox committed:
Revert "Remove inversemousearea tests which test nothing"

This reverts commit 3ebcc600fbd07e16e3eaf961a6ac055d96665ede.

UniversalSuperBox committed:
Skip failing inversemousearea tests

mymike00 commented:
Yeah, how to solve them? just trigger again the build?

dobey committed:
Fix the alarm tests.

dobey committed:
Avoid a possible null object check.

dobey committed:
Fix the documentation for Qt 5.9.

dobey committed:
Remove tests that no longer fall here.

dobey committed:
Set id on the qml view we create.

dobey committed:
Re-enable skipped inversemousearea tests.

dobey committed:
Use currentDateTime for order tests.

dobey committed:
Update units tests.

dobey committed:
Update the custom_qpa.

dobey committed:
Update swipearea test.

dobey committed:
Remove contenthub test and re-enable others.

As contenthub is a separate repository, this test is no longer valid here.
With some of the other fixes, these other tests can be re-enabled now.

dobey committed:
Disable pep8 until autopilot can be re-enabled.

dobey committed:
Remove gles conversion hack.

dobey committed:
Create a dummy touch device for tests.

dobey committed:
Replace waitForEvents with processEvents.

This should resolve #12 when merged to the bionic branch.

dobey committed:
Merge branch 'xenial_-_fix-failing-uitk-tests' into bionic_-_fix-failing-uitk-tests

dobey committed:
Remove left-behind endif from merge.

dobey committed:
Another leftover ifdef fix.

dobey committed:
Try to build without parallel tests.

dobey committed:
Revert parallel removal.

dobey committed:
Fix warning in popup.

dobey committed:
Fix warning in scrollbar style.

dobey committed:
Fix binding loop in Picker.

dobey committed:
Update expected warnings list.

dobey committed:
Qt 5.9 uses nullptr.

dobey committed:
Fix abstractbutton 1.3 tests.

dobey committed:
Skip tests neding OpenGL context.

dobey committed:
Make tests pass on amd64.

dobey committed:
Don't force touch gesture to run first.

dobey committed:
Disable tests on armhf for now.

dobey committed:
Pull in test fix changes from bionic branch.

dobey committed:
Must diable arm64 on xenial too.

UniversalSuperBox committed:
Merge pull request #31

Make the builds pass again on xenial

UniversalSuperBox commented:
Thanks to https://github.com/ubports/ubuntu-ui-toolkit/pull/30 and https://github.com/ubports/ubuntu-ui-toolkit/pull/31, we finally have everything in place to build the UITK again and PRs can be merged.

UniversalSuperBox reviewed (changes requested):
@NeoTheThird brought up a good point with this PR: we would need to test many apps with it before we merge. This could break the design of any app which uses the UbuntuShape, even relatively small apps like [8192](https://open-store.io/app/8192.neothethird).

An ideal test plan would be:

1. Wait for a good time in the `xenial_-_edge` cycle (now is not a good time)
2. Merge this into edge and encourage testers to try out all their apps

Alternatively, you can get people to install this PR (`sudo ubports-qa install PR-22`) and test out a lot of apps to see if any break, then report the breakage to the authors. This might be the faster solution.

Let me know what you think.

mymike00 commented:
I'd prefer to ask in the QA team/main supergroups for tester to try this PR, imho more people can test it and it'd faster.

cibersheep commented:
Ouh. Charging my testing device :)

bhdouglass committed:
Made pull to refresh use translations for the ubuntu-ui-toolkit domain

cibersheep commented:
I have tried to install this but I can't see any change in the ui. Might it be a qrc (qml cached file?)

mymike00 commented:
@cibersheep Mmh that's odd. when I installed it with crossbuilder I see them after the first reboot iirc. I tried now, after an update, with `ubports-qa` but I cannot make it working...
```
$ sudo ubports-qa install ubuntu-ui-toolkit 22
CRITICAL:ubports-qa:Jenkins build not found
```

UniversalSuperBox commented:
You might need to do `ubports/reponame` for some reason.

mymike00 committed:
Merge remote-tracking branch 'mymike/xenial_ubuntushape' into xenial_ubuntushape

mymike00 commented:
I noticed that with colored ubuntu shapes you don't notice too much the drop shadows while it gave a better look than the inset shadow imho.
yeah, I agree that the option selector isn't that great but if we make it with the inset shadow it would be the only component different from others...
but, all in all, this pr doesn't introduce so drastic changes therefore I think it could be merged and then if a lot of people say to prefer some components with the old look then we can fix it later
remember that there's also another ubuntu shape aspect option, which is flat, with no shadow at all. it might be nicer for the option selector...

dobey commented:
> remember that there's also another ubuntu shape aspect option, which is flat, with no shadow at all. it might be nicer for the option selector...

I think flat would probably look better for all the cases. Only the toggle switches might be a little odd, but it should already be specifying inset and outset for the shadow types of its elements, I would think.

mymike00 commented:
No, I don't think they do...
to decide the default aspect idk, it'd prefer dropshadow because when you use pure ubuntuShape for (usually) images/logo it fits better while other uitk components could have a specific aspect for their selves...
I'll may do some tests and see what looks nicer

cibersheep commented:
I'm also a fan of flat. But I wasn't able to test this PR yet. Can you believe that?

mymike00 commented:
I wasn't able to install it with ubports-qa iirc, only with crossbuilder...

mymike00 commented:
so I tested with default aspect `Flat` and `Switch` and `OptionSelector` doesn't look nice...
we have to decide if we prefer them `Inset` or `DropShadow`, let me know
![immagine](https://user-images.githubusercontent.com/28359593/56079738-8ff54800-5df8-11e9-8aa6-de94f09d98d2.png)
![immagine](https://user-images.githubusercontent.com/28359593/56079742-9a174680-5df8-11e9-80c6-99013d415e62.png)
![immagine](https://user-images.githubusercontent.com/28359593/56079745-9d123700-5df8-11e9-813a-48c1bf79e2fe.png)

mymike00 commented:
what are we going to do with this?
the thing is that I don't like the Inset shadow by default (but looks good in some component like OptionSelector, Switch, CheckBox, etc...). the other 2 choices are Flat or DropShadows: usually UbuntuShape is used for images, app icons, buttons ... that have colored bg. in those situation (quite always) I think it suits better the DropShadows so I'd make it the default.
are you ok?

cibersheep commented:
Sounds ok :)

mymike00 committed:
some components with Inset aspect by default

OptionSelector, Switch, CheckBox

mymike00 commented:
Now `OptionSelector`, `Switch` and `CheckBox` use bt default `Inset` aspect (which is their default look now, before this PR) while the default for everything else is DropShadow

UniversalSuperBox commented:
I've reconfigured the CI so pull requests will build now. It'll be a little while until yours is finished, Jenkins is quite upset with me for that.

This PR has become quite littered with commits from `xenial`. If I can ask it of you, could you create a new PR built off of the current `xenial` branch with only your changes added? That will ensure we're only viewing and testing your changes.

mymike00 commented:
Sure, i'm gonna do it this week
